core/src/main/scala/org/apache/spark/rpc/netty/Inbox.scala

```diff
@@ -0,0 +1,258 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.rpc.netty
+
+import java.util.concurrent.ConcurrentLinkedQueue
+
+import scala.util.control.NonFatal
+
+import org.apache.spark.{SparkException, Logging}
+import org.apache.spark.rpc.{ThreadSafeRpcEndpoint, RpcAddress, RpcEndpoint}
+
+private[netty] sealed trait InboxMessage
+
+private[netty] case class ContentMessage(
+    senderAddress: RpcAddress,
+    content: Any,
+    needReply: Boolean,
+    context: NettyRpcCallContext) extends InboxMessage
+
+/**
+ * A message type that will be posted to all registered [[RpcEndpoint]]
+ */
+private[netty] sealed trait BroadcastMessage extends InboxMessage
+
+private[netty] case object OnStart extends InboxMessage
+
+private[netty] case object OnStop extends InboxMessage
+
+private[netty] case class Associated(remoteAddress: RpcAddress) extends BroadcastMessage
+
+/**
+ * A broadcast message that indicates
+ */
+private[netty] case class Disassociated(remoteAddress: RpcAddress) extends BroadcastMessage
+
+/**
+ * A broadcast message that indicates a network error
+ */
+private[netty] case class AssociationError(cause: Throwable, remoteAddress: RpcAddress)
+  extends BroadcastMessage
+
+private[netty] abstract class Inbox(
+    val endpointRef: NettyRpcEndpointRef, val endpoint: RpcEndpoint) extends Logging {
+
+  protected val messages = new ConcurrentLinkedQueue[InboxMessage]()
+
+  /**
+   * Process stored messages. Return `true` if the `Inbox` is already stopped, and the caller will
+   * release all resources used by the `Inbox`.
+   */
+  def process(dispatcher: Dispatcher): Boolean
+
+  def post(message: InboxMessage): Unit
+
+  protected def onDrop(message: Any): Unit = {
+    logWarning(s"Drop ${message} because $endpointRef is stopped")
+  }
+
+  def stop(): Unit
+
+  protected def safelyCall(endpoint: RpcEndpoint)(action: => Unit): Unit = {
+    try {
+      action
+    } catch {
+      case NonFatal(e) => {
+        try {
+          endpoint.onError(e)
+        } catch {
+          case NonFatal(e) => logError(s"Ignore error: ${e.getMessage}", e)
```

Comment:
`logError` seems too much for an exception that you're intentionally ignoring. Also `${e.getMessage}` is redundant since you're printing the whole exception.